packages/odyssey-react-mui/src/labs/DataComponents/useFilterConversion.ts

```diff
@@ -0,0 +1,92 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { useCallback, useMemo } from "react";
+import { MRT_RowData } from "material-react-table";
+
+import { DataFilter } from "../DataFilters";
+import { DataTableColumn } from "../../DataTable";
+import { UniversalProps, TableProps } from "./componentTypes";
+
+type FilterConversionType = {
+  columns?: TableProps["columns"];
+  filters?: UniversalProps["filters"];
+};
+
+export const useFilterConversion = ({
+  columns,
+  filters,
+}: FilterConversionType) => {
+  const convertFilterSelectOptions = useCallback(
+    (options: DataTableColumn<MRT_RowData>["filterSelectOptions"]) =>
+      options?.map((option) =>
+        typeof option === "string"
+          ? {
+              label: option,
+              value: option,
+            }
+          : {
+              // If the option isn't a string, it must have value and/or option defined
+              // If either is undefined, use the other
+              label: option.label ?? option.value,
+              value: option.value ?? option.label,
+            },
+      ),
+    [],
+  );
+
+  const convertColumnToFilter = useCallback(
+    (column: DataTableColumn<MRT_RowData>) =>
+      column.enableColumnFilter !== false && column.accessorKey
+        ? ({
+            id: column.accessorKey,
+            label: column.header,
+            variant: column.filterVariant,
+            options: convertFilterSelectOptions(column.filterSelectOptions),
+          } satisfies DataFilter as DataFilter)
+        : null,
+    [convertFilterSelectOptions],
+  );
+
+  const dataTableFilters = useMemo(() => {
+    const providedFilters = filters || columns;
+    if (!providedFilters) {
+      return [];
+    }
+    return providedFilters.reduce<DataFilter[]>((accumulator, item) => {
+      if (typeof item === "string") {
+        const foundColumn = columns?.find(
+          (column) => column.accessorKey === item,
+        );
+        if (foundColumn) {
+          const filter = convertColumnToFilter(foundColumn);
+          if (filter) {
+            return accumulator.concat(filter);
+          }
+        }
+      } else if ("accessorKey" in item) {
+        // Checks if it's a column
+        const filter = convertColumnToFilter(item);
+        if (filter) {
+          return accumulator.concat(filter);
+        }
+      } else if ("label" in item) {
+        // Checks if it's a DataFilter
+        return accumulator.concat(item);
+      }
+      // If none of the conditions match, item is ignored (not mapping to undefined)
+      return accumulator;
+    }, []);
+  }, [columns, filters]);
+
+  return dataTableFilters;
+};
```

Comment:
We should probably revisit this code and see if there's any way to make it simpler. There are a lot of `if` checks here that I hope we can reduce somehow with an object or some other function.